frontend/src/pages/FAQPage.jsx

```diff
@@ -1,99 +1,206 @@
+                placeholder="Post content"
+                value={newPost.content}
+                onChange={(e) => setNewPost({ ...newPost, content: e.target.value })}
+                required
+              />
+            </div>
+            <button
+              type="submit"
+              className="mt-4 px-6 py-3 bg-indigo-600 text-white font-semibold rounded-lg shadow-md hover:bg-indigo-700 transition duration-300 ease-in-out"
+            >
+              Submit Post
+            </button>
+          </form>
+          <section>
+            <div className="flex justify-between items-center mb-6">
+              <h2 className="text-4xl font-semibold text-indigo-600">All Posts</h2>
+              <button
+                className="px-4 py-2 bg-indigo-500 text-white font-semibold rounded-lg shadow-md hover:bg-indigo-600 transition duration-300"
+                onClick={handleSort}
+              >
+                Sort
+              </button>
+            </div>
 
-
-const FAQItem = ({ question, answer }) => {
-  const [isOpen, setIsOpen] = useState(false);
-
-  const toggleFAQ = () => setIsOpen(!isOpen);
-
-  return (
-    <div className="border-b border-gray-300 py-4">
-      <div
-        className="flex justify-between items-center cursor-pointer"
-        onClick={toggleFAQ} role='button'
-      >
-        <h3 className="text-lg font-medium text-gray-800 dark:text-white">{question}</h3>
-        <span className="text-gray-500">{isOpen ? '-' : '+'}</span>
+            <div className="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 gap-8">
+              {posts.map((post, index) => (
+                <div key={post.id} className={`relative w-full perspective transition-transform duration-300 hover:scale-105`}>
+                  <div className="relative w-full h-64 transform-style-3d transition-transform duration-500 ease-in-out">
+                    <div className="absolute inset-0 p-4 bg-indigo-300 border border-indigo-500 rounded-lg">
+                      <h3 className="text-xl font-semibold text-indigo-700 mb-4">{post.title}</h3>
+                      <p className={`text-white ${expandedPosts.includes(post.id) ? 'line-clamp-none' : 'line-clamp-3'}`}>
+                        {post.content}
+                      </p>
+                    </div>
+                    <button
+                      className="absolute bottom-2 right-2 text-sm font-semibold text-indigo-700 underline focus:outline-none"
+                      onClick={() => toggleExpand(post.id)}
+                    >
+                      {expandedPosts.includes(post.id) ? 'Collapse' : 'Read More'}
+                    </button>
+                  </div>
+                </div>
+              ))}
+            </div>
+          </section>
+        </section>
       </div>
-      {isOpen && <p className="mt-2 text-gray-600">{answer}</p>}
-    </div>
+
+      <Footer />
```

Comment:
## dont use footer 
This will create footer container twice, 
see preview : https://decentrade-git-fork-shariq2003-e-3d36b2-adarsh-mauryas-projects.vercel.app